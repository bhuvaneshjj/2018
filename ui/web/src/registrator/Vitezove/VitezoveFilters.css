.VitezoveFilters {
  align-items: center;
  display: flex;
  justify-content: flex-start;
}

.VitezoveFilters__kategorie {
  margin-left: 2em;
}
