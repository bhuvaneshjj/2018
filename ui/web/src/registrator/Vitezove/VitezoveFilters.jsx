import React from 'react';
import PropTypes from 'prop-types';
import { ButtonGroup } from 'react-bootstrap';
import KategorieFilter from '../Filterable/KategorieFilter';
import KategorieSubFilter from '../Filterable/KategorieSubFilter';
import './VitezoveFilters.css';

const VitezoveSubFilters = ({ kategorieSubFilters }) => {
  const { length } = kategorieSubFilters;
  if (length >= 6) {
    const half = length / 2;
    return (
      <div className="VitezoveFilters__sub-kategorie">
        <div>
          {kategorieSubFilters.slice(0, half).map(({ id, ...props }) => (
            <KategorieSubFilter key={id} {...props} />
          ))}
        </div>
        <div>
          {kategorieSubFilters.slice(half, length).map(({ id, ...props }) => (
            <KategorieSubFilter key={id} {...props} />
          ))}
        </div>
      </div>
    );
  }
  return (
    <div className="VitezoveFilters__sub-kategorie">
      {kategorieSubFilters.map(({ id, ...props }) => (
        <KategorieSubFilter key={id} {...props} />
      ))}
    </div>
  );
};

VitezoveSubFilters.propTypes = {
  kategorieSubFilters: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      id: PropTypes.string.isRequired,
      typ: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired
    }).isRequired
  ).isRequired
};

const VitezoveFilters = ({ kategorieFilters, kategorieSubFilters, kategorieSubFiltersVisible }) => (
  <div className="VitezoveFilters">
    <ButtonGroup className="VitezoveFilters__kategorie">
      {kategorieFilters.map(({ active, typ, onClick }) => (
        <KategorieFilter active={active} key={typ} typKategorie={typ} onClick={onClick} />
      ))}
    </ButtonGroup>

    {kategorieSubFiltersVisible && <VitezoveSubFilters kategorieSubFilters={kategorieSubFilters} />}
  </div>
);

VitezoveFilters.propTypes = {
  kategorieFilters: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      typ: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired
    }).isRequired
  ).isRequired,
  kategorieSubFilters: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      id: PropTypes.string.isRequired,
      typ: PropTypes.string.isRequired,
      zkratka: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired
    }).isRequired
  ).isRequired,
  kategorieSubFiltersVisible: PropTypes.bool.isRequired
};

export default VitezoveFilters;
