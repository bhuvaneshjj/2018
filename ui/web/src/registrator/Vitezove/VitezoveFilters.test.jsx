import React from 'react';
import renderer from 'react-test-renderer';
import VitezoveFilters from './VitezoveFilters';

const kategorieFilters = [
  { active: true, typ: 'maraton', onClick: jest.fn() },
  { typ: 'cyklo', onClick: jest.fn() },
];
const kategorieSubFilters = [
  { id: '1', active: true, typ: 'maraton', zkratka: '1M', onClick: jest.fn() },
  { id: '2', typ: 'maraton', zkratka: '2M', onClick: jest.fn() },
  { id: '3', typ: 'maraton', zkratka: '3M', onClick: jest.fn() },
];

it('renders kategorie filters', () => {
  const component = renderer.create(
    <VitezoveFilters
      kategorieFilters={kategorieFilters}
      kategorieSubFilters={[]}
      kategorieSubFiltersVisible={false}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

it('renders kategorie subFilters', () => {
  const component = renderer.create(
    <VitezoveFilters
      kategorieFilters={[kategorieFilters[0]]}
      kategorieSubFilters={kategorieSubFilters}
      kategorieSubFiltersVisible={true}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
