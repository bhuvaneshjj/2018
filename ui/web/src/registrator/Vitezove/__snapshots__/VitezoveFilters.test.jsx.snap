// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders kategorie filters 1`] = `
<div
  className="VitezoveFilters"
>
  <div
    className="VitezoveFilters__kategorie btn-group"
  >
    <span
      className="KategorieFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieFilter--maraton--active btn btn-default active"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "100%",
            }
          }
        >
          <img
            alt="maraton"
            height={30}
            src="maraton.png"
            title="maraton"
            width={30}
          />
           
          maraton
        </span>
      </button>
    </span>
    <span
      className="KategorieFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieFilter--cyklo--inactive btn btn-default"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "100%",
            }
          }
        >
          <img
            alt="cyklo"
            height={30}
            src="cyklo.png"
            title="cyklo"
            width={30}
          />
           
          cyklo
        </span>
      </button>
    </span>
  </div>
</div>
`;

exports[`renders kategorie subFilters 1`] = `
<div
  className="VitezoveFilters"
>
  <div
    className="VitezoveFilters__kategorie btn-group"
  >
    <span
      className="KategorieFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieFilter--maraton--active btn btn-default active"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "100%",
            }
          }
        >
          <img
            alt="maraton"
            height={30}
            src="maraton.png"
            title="maraton"
            width={30}
          />
           
          maraton
        </span>
      </button>
    </span>
  </div>
  <div
    className="VitezoveFilters__sub-kategorie"
  >
    <span
      className="KategorieSubFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieSubFilter--maraton--active btn btn-default active"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "85%",
            }
          }
        >
           (1M)
        </span>
      </button>
    </span>
    <span
      className="KategorieSubFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieSubFilter--maraton--inactive btn btn-default"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "85%",
            }
          }
        >
           (2M)
        </span>
      </button>
    </span>
    <span
      className="KategorieSubFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieSubFilter--maraton--inactive btn btn-default"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "85%",
            }
          }
        >
           (3M)
        </span>
      </button>
    </span>
  </div>
</div>
`;
